### ISSUE TITLE
Portal Storms: Future Plans and Feedback

### ISSUE THREAD

**Issue by Ramza13** [CleverRaven/Cataclysm-DDA#52475]
# Portal Storms Future
Portal storms are a feature in progress and not done yet. I have opened this to show some future planned ideas and to gather feedback from people.  All of the ideas listed here are things I am considering doing, not set in stone, if you like or dislike some please let me know or add your own suggestions.

# New portal storm effects:
- NPC spawn, a human from a similar universe ends up here.  Potential ally with portal storm lore, only small chance of being spy/puppet for vast nether intelligences.
- Spawn fields other than smoke, there are a lot of field types and variables that could be interesting.

# Loot:
Portal storms could use a carrot to encourage exploring them.  
- Items that help you know when a portal storm is approaching.
- Ways to lower the storms base strength later in the game.
- Artifacts might make sense here.
- Maybe portal storm based mutations you could get

# New enemies:
Ideas for new enemies spawning, design goal to make things that aren't just hard to fight but harm you in other ways.
- Enemy that increase strength of portal storm but doesn't harm you.
- Enemy that drains calories away from you with every blow.
- Enemy that spawns rarely outside of portal storms.  It will chase the player down and its only attack will kill itself with out hurting you.  If it dies a portal storm happens.

# Achievements:
Right now there are no achievements for portal storms at all. 
- Surviving your first storm, 10th storm, 25th, etc
- Sleeping through a portal storm.
- Getting hit with some of the higher end effects like incorporeal for the first time.

# New type of status:
This idea still needs some work and probably won't make the 0.G release.
Right now portal storm enemies mostly do non damage type things like make you tired, I want to add another effect that I haven't settled on a name for.  Maybe negative energy or reality damage or something like that.  The idea being that contact with nether enemies actually damages you in fundamental ways as they steal bits of the reality that make you up.  It would be a variable hidden from the player except possibly with a vague idea of if its small or large.  Portal storm enemies would increase it.  

Once in a while (not just during portal storms)  if you have enough negative energy some will be used to cause a random anomaly.  These anomalies would usually be bad but not always.  For example a spatial anomaly forms near you and causes projectiles to miss, both yours and enemies.  Or some of what are currently portal storm effects could happen like minor teleportation.  Maybe you get tired all of a sudden or take a small amount of damage as your altered physics harm your body.  The more negative energy you have the bigger and more often the effects.

The other part would be that it becomes possible to barter with some nether creatures or voices.  What they could offer you is more targeted anomaly creation.  Maybe you can choose when to summon a projectile altering one or if one happens it only affects enemies.  The goal here would be to make negative energy more useful but still not something you can totally control.  There is always a risk of bad effects involved.  Alternately they can give you bonuses that only apply during portal storms.  At higher levels portal storms might be on balance actually helpful and something you'd like to cause.  The cost for these bargains is gaining negative energy and huge amounts of fatigue and/or incorporeal.  Or possibly some missions to complete if I can think of something cool a nether entity would want.  Maybe doing something that strengthens portal storms or killing some of its rivals.

Also these bargains are not risk free.  Not all the voices are telling you the truth and even the ones that are, ultimately want to drain your energy and/or pull you into the nether( a process you might survive though not in a manner current you would call life and a non standard game over).  The idea here is to get a little bit of the energy of gods from nethack or dungeon crawl stone soup, possibly having multiple different entities to bargain with. Though not quite as predictable as those gods.  

# Portal Storm dungeon
Portal storms create buildings sometimes, these buildings vanish at the end of the storm.  Possibly you get a mission to find them alerting you of their location, the mission goes away when the dungeon does.  The goal is to make this a small dungeon, the difficulty is that the enemies and even the terrain can teleport you away.  Their might be loot or their might be an alter of sorts which you can use to bargain with nether entities( this probably wouldn't be the only way but would be safer or more rewarding.)

# Balance:
Here is where I need feedback the most.  Its likely the rate portal storms happen, how long they last, and how fast their strength increases could use some finetuning.  I'm considering increasing how often they occur but will wait for more playtesting.
- Make giant appendages hard but killable. Been suggested several times.

**Comment by Ramza13** [comment]
> 
> 
> Fun, how are you planning to spawn in monsters? I'd like to support this with DinoMod if possible.

I'm planning to add new monsters spawns but thats just json. Right now you can look at portal_effect_on_conditions.json/check out NPCs.md and see how its done right now.  If you have any questions feel free to ask me here or on the discord

**Comment by Light-Wave** [comment]
Right now, the way alien grass doesn't always despawn is rather annoying.

- A portal storm happens when I'm outside
- I take the shortest route back home
- Alien grass spawns
- I continue home as quickly as I can
- There is not permanent alien grass blocking the shortest routes between my base and other important locations, messing with auto travel

**Comment by LyleSY** [comment]
It might be interesting to interact with a past, future, parallel, or fake you that mirrors or takes the opposite of the player's actions in the conversation

**Comment by LeahLuong** [comment]
It might not be interesting to interact with a past, future, parallel, or fake you that mirrors or takes the opposite of the player's actions in the conversation

**Comment by Ramza13** [comment]
> Right now, the way alien grass doesn't always despawn is rather annoying.
> 
> * A portal storm happens when I'm outside
> * I take the shortest route back home
> * Alien grass spawns
> * I continue home as quickly as I can
> * There is not permanent alien grass blocking the shortest routes between my base and other important locations, messing with auto travel

Thanks for pointing this out.  I see what is happening there and will write a fix.

**Comment by Ramza13** [comment]
Updated my plans to add a new portal storm idea I am still working on, negative energy/bargains with nether entities.

**Comment by thethunderhawk** [comment]
IIRC, lorewise, the “dimensional heuristics” that allow for AI in our universe are affected significantly by how “thin” the barrier between universes is. When the cataclysm hit, it caused Melchior and other large AI to become much more capable. So, maybe portal storms cause bots to behave erratically?

**Comment by Terrorforge** [comment]
I'm about six months deep in a game right now, and the portal storms are getting pretty vicious. I figured I'd offer some feedback:

They always catch me off guard. Even though they're now quite frequent, I keep forgetting they're a thing, and I never notice any of the warnings until the weather actually changes. And at this point, they hit so hard and so fast that can cause real problems. For example, the other day I was sitting in my (roofless) APC next to a radio tower when a portal storm hit. I sprinted out to get to radio shack, had [four flaming eyes spawn](https://cdn.discordapp.com/attachments/699896059555610717/928050308184166491/unknown.png) and instantly got a "badly tainted mind". I survived that, and so far it hasn't screwed me completely, but it does _feel_ kind of bullshit to just instantly get slapped with the nasty stuff.

I'm not sure how I feel about the "incorporeal" effect. In principle, I like it. It's interesting that it's not a complete negative (prime candidate for a bargain reward), and I like being made to survive under novel conditions - but I say that as a lategame mutant who's perfectly comfortable fighting unarmed and unarmored. If I was in any way dependent on gear, dropping it would be a death sentence. It's also just kind of annoying to have to go and put everything back on, especially since it can mess with hotkeys and things like that. And if you're driving when it happens, _boy_.

And yeah, actually getting through them is just kind of boring. I've had a lot of great moments where a portal storm starts and I'm desperately trying to get to shelter in time, making bad decisions, fighting too many zombies and feeling a powerful sense of relief when I limp in under a roof, backpack in tatters and knife half broken - and then I just plug my ears and wait for the storm to pass. I think you have the right idea in that there need to be activities you can only do during portal storms. Perhaps even some that don't require going into the storm itself? Maybe you can only craft certain things during a storm, for example.

Oh, and I have a question: are you supposed to be safe from portal storms in a car? I seem to have gotten the idea in my head that you are, but I can't find any reference to it. I'm definitely not, but I had to tear the roof off because I'm Huge, so that's not saying much.

**Comment by Ramza13** [comment]
> I'm about six months deep in a game right now, and the portal storms are getting pretty vicious. I figured I'd offer some feedback:
> 
> They always catch me off guard. Even though they're now quite frequent, I keep forgetting they're a thing, and I never notice any of the warnings until the weather actually changes. And at this point, they hit so hard and so fast that can cause real problems. For example, the other day I was sitting in my (roofless) APC next to a radio tower when a portal storm hit. I sprinted out to get to radio shack, had [four flaming eyes spawn](https://cdn.discordapp.com/attachments/699896059555610717/928050308184166491/unknown.png) and instantly got a "badly tainted mind". I survived that, and so far it hasn't screwed me completely, but it does _feel_ kind of bullshit to just instantly get slapped with the nasty stuff.
> 
> I'm not sure how I feel about the "incorporeal" effect. In principle, I like it. It's interesting that it's not a complete negative (prime candidate for a bargain reward), and I like being made to survive under novel conditions - but I say that as a lategame mutant who's perfectly comfortable fighting unarmed and unarmored. If I was in any way dependent on gear, dropping it would be a death sentence. It's also just kind of annoying to have to go and put everything back on, especially since it can mess with hotkeys and things like that. And if you're driving when it happens, _boy_.
> 
> And yeah, actually getting through them is just kind of boring. I've had a lot of great moments where a portal storm starts and I'm desperately trying to get to shelter in time, making bad decisions, fighting too many zombies and feeling a powerful sense of relief when I limp in under a roof, backpack in tatters and knife half broken - and then I just plug my ears and wait for the storm to pass. I think you have the right idea in that there need to be activities you can only do during portal storms. Perhaps even some that don't require going into the storm itself? Maybe you can only craft certain things during a storm, for example.
> 
> Oh, and I have a question: are you supposed to be safe from portal storms in a car? I seem to have gotten the idea in my head that you are, but I can't find any reference to it. I'm definitely not, but I had to tear the roof off because I'm Huge, so that's not saying much.

So thank you very much for your feedback!  I try to playtest this stuff but I rarely survive a week let alone a month so a lot of it I have to kinda guess at.  

The idea of certain crafts only being possible during a storm is a great one, especially if some of the ingredients  are nether fauna or flora.  Not sure what the hell skill that would be though.

For warnings do you have ideas to make them better?  My thoughts are I could make them a popup but that might be obnoxious or I could add small effects to them like a weak monster spawn or brief sun vanishing but you could easily miss those too.  At some point an item or mutation that gives you better warnings might make sense as well.

You should be safe from a portal storm while inside, the code has logic for if vehicles counts as indoors or not based I think on if there is a roof or not.  So in your case you wouldn't be safe.  That said I might add checks to the teleportation stuff and incorporeal to prevent it happening while driving because that's just ugly.

I agree that the incorporeal is pretty harsh and want to move that out of a random effect and into a monster attack or effect of making a bargain with a nether being.  Actually eventually I want to do that with a bunch of the effects so you have at least a chance to avoid them.

Thanks again and your feedback is greatly appreciated!

**Comment by Terrorforge** [comment]
> The idea of certain crafts only being possible during a storm is a great one, especially if some of the ingredients are nether fauna or flora. Not sure what the hell skill that would be though.

It wouldn't have to be just one skill, it could depend on the craft in question. But if we're talking about improvising uses for weird alien flora and fauna, Survival does jump to mind.

> For warnings do you have ideas to make them better? My thoughts are I could make them a popup but that might be obnoxious or I could add small effects to them like a weak monster spawn or brief sun vanishing but you could easily miss those too. At some point an item or mutation that gives you better warnings might make sense as well.

I think it has to be a popup. Crucially, it has to interrupt long-term activities like crafting, or you'll still miss it half the time. There's currently an "oh no! a portal storm" message when one starts, right? It's not any worse than that. You could even replace it with the earlier warning, but I think it's fine to have both.

I do kind of like the idea that a full portal storm would be preceded by some kind of "Gathering portal storm" weather that's weird but mostly safe. Perhaps it could even give access to some of the lower-level rewards that would be available during a full storm.

> You should be safe from a portal storm while inside, the code has logic for if vehicles counts as indoors or not based I think on if there is a roof or not. So in your case you wouldn't be safe. That said I might add checks to the teleportation stuff and incorporeal to prevent it happening while driving because that's just ugly.

I feel like I've had issues with being affected even when I was in a car, but since I didn't really know what the conditions for safety were supposed to be, I didn't really troubleshoot. I'll keep an eye out for it in future runs.

I did have a moment where I was teleported out of my car. Threw a bunch of errors, but the car stopped pretty quickly and I had a nice little jog to the nearest building. I presumably could have been run over, though, which again seems a bit harsh - especially since "hop in your car and put the pedal to the metal" is the natural response to getting caught out in a portal storm.

**Comment by Terrorforge** [comment]
These new messages are an improvement. I'm sure not going to miss another portal storm. But I see two immediate issues with the implementation:

1. They don't seem to give you the option to cancel your current activity, so it's still pretty easy to get caught out and lose precious time.
2. You get a pop-up for every single message, and they don't stop even when the actual portal storm begins. When I said it wouldn't be so bad I was envisioning one pop-up when the warnings begin, and maybe another when the portal storm proper begins and/or ends. But I got like a dozen over the course the course of my latest storm, and that did in fact get a bit annoying.

**Comment by Zeropol** [comment]
The messages about the portal storms show up even when sleeping.
![image](https://user-images.githubusercontent.com/28099824/148808302-528fc0b0-2fa4-4126-8e0a-b32f94d36e4d.png)

**Comment by AcidAntOnAMinefield** [comment]
I commend the creativity behind the feature but ultimately I still think this particular weather effect should be an option for players to choose whether they want it in their worlds or not. 

Personally:
- I think it's too spammy in nature and it's essentially an "oh, oops you lose your progress/die now" button. The moment a storm occurs, the player gets swarmed by creatures that essentially instantly reduce whatever clothes they're wearing to shreds (or outright kill them with fatigue+pain stacks). That or the player gets the pleasant surprise spawns from Unseen Hunters, Flaming Eyes or Hounds of Tindalos, creatures that surely are perfectly balanced themselves;
- Adding something that forces players out of the only place that's safe is essentially telling them "you're playing the game the way *I* want you to play", which I find it to be a rather hostile approach, as it essentially disallows players to experience the game at their own pace.
- The random effects suggestions (the missile missing thing, the random "getting tired" thing, etc) and the anomalies also do not sound like a good idea to me. There was a time when things like these were deemed "unrealistic" and would be relegated to mod status and not mainline, and honestly, the whole thing *is* not realistic, it goes against the whole "realism" idea the game has been steering towards. 

I don't know. It was a fun thing when it first happened, for its novelty but it got frustrating and annoying after a while. The fact that my character has to choose between essentially drop everything that he/she is doing or just die "because reasons" got really old, really fast.

**Comment by Ramza13** [comment]
> I commend the creativity behind the feature but ultimately I still think this particular weather effect should be an option for players to choose whether they want it in their worlds or not.
> 
> Personally:
> 
> * I think it's too spammy in nature and it's essentially an "oh, oops you lose your progress/die now" button. The moment a storm occurs, the player gets swarmed by creatures that essentially instantly reduce whatever clothes they're wearing to shreds (or outright kill them with fatigue+pain stacks). That or the player gets the pleasant surprise spawns from Unseen Hunters, Flaming Eyes or Hounds of Tindalos, creatures that surely are perfectly balanced themselves;
> * Adding something that forces players out of the only place that's safe is essentially telling them "you're playing the game the way _I_ want you to play", which I find it to be a rather hostile approach, as it essentially disallows players to experience the game at their own pace.
> * The random effects suggestions (the missile missing thing, the random "getting tired" thing, etc) and the anomalies also do not sound like a good idea to me. There was a time when things like these were deemed "unrealistic" and would be relegated to mod status and not mainline, and honestly, the whole thing _is_ not realistic, it goes against the whole "realism" idea the game has been steering towards.
> 
> I don't know. It was a fun thing when it first happened, for its novelty but it got frustrating and annoying after a while. The fact that my character has to choose between essentially drop everything that he/she is doing or just die "because reasons" got really old, really fast.

Thanks for your feedback.  To help me better understand it here are a few questions.  

For your first point, part of the idea is that you should have at least some warning to get to shelter, are these not enough or too easy to miss?
For your second point, I'm not sure what you are talking about, at the moment if you are inside of a house or covered vehicle you should be completely safe.  Are you referring to possible planned features or is there something I am missing?

I think your concerns are valid and I probably made portal storms too frequent to help out testing which makes it worse.  As for them being a mod I can understand that perspective and made a portal storm disabling mod that was rejected as the current goal is to make them fun so you won't want to disable them.  That said it would take you 30 seconds to remove them via json(just find and remove EOC_PORTAL_STORM_WARN_OR_CAUSE_RECURRING) and as they become a more polished feature I do think making them a bit more customizable via scenarios or options or something would be nice.

**Comment by AcidAntOnAMinefield** [comment]
I wouldn't say they are easy to miss, at least now compared to its early stages. The messages and the hints are appreciated, although in situations where the chat log goes crazy with other little events (eating, taking pills, listening to music, hauling items, fighting, etc), some subtle hints can be missed. Although I believe that may have been mitigated by the big "center screen" messages that show up now. 
The main gripe I have is that on occasion you just get one or two creatures which is fair, but sometimes you can get absolute bad luck and get yourself surrounded (sometimes literally) by creatures with no means to escape, especially if those creatures are "special" in some way that prevents you from escaping altogether (being Blinded by a Flaming Eye and them surrounded by a Hound of Tindalos and its mirrors), all the while being "fatigued" and undressed by two types of Portal Storm unique creatures can be rather unpleasant and, to an extent, unfair.

I should've been clearer and quoted what I meant with the second point, but yes I was referring to the potential/planned new features (the creature that pulls you out of your shelter, or the effects "splashing" into shelters). 

Although I should add that I do understand why they are frequent, it was essentially the same thing with the Trans Coast Logistics POI - pros and cons of playing experimental :) - and even for someone with no technical know-how in terms of coding, the "best" work-around for me when I just want to play and don't worry about something that may/may not prevent me from doing something is to just debug it away (usually via the weather debug option). 
Although I will look into what you mentioned above and try to make a little mod for my specific little world. I highly appreciate the guidance there.

I'll still be looking forward to see how this develops because, despite what I mentioned about the "realism" aspect the game has adopted over the years to the detriment of other features, I still maintain that "unrealistic" things have their place in the game - some that were removed, others that were rejected, etc - and I think weather effects definitely have a ton of potential to make the game more interesting (be it the portal storms, acid rains, solar flares, etc).

**Comment by Terrorforge** [comment]
The pendulum seems to swing back and forth on "realism". It's always a guiding principle, of course, but with things like the Exodii and portal storms, it seems we're currently swinging away from it somewhat.

> Adding something that forces players out of the only place that's safe is essentially telling them "you're playing the game the way I want you to play", which I find it to be a rather hostile approach, as it essentially disallows players to experience the game at their own pace.

On one hand, that's kind of the point? A big part of the purpose of portal storms is to be a danger that imposes itself on the player, rather than something the player seeks out, because the game can become a bit routine if you always get to choose your battles.

But on the other, we don't want to badger players. That's just annoying. Initially I thought I would like something like a grappler monster forcing you outside, but now I vastly prefer the various suggestions that give players _incentive_ to venture into portal storms rather than forcing it.

**Comment by Anandar83** [comment]
Got a couple of small ideas, portal storms could affect the battery charge of tools and electric cars, and for diesel /gasoline cars could cause them to stall or fail to start, also with mp3 players, radios and stereo systems you could hear nether creatures talking to you, making 2 way radios a good way to get quests from said nether creatures.

**Comment by drhead** [comment]
Interesting feature.  STALKER-style blowouts seem like they'd fit very well with the game.

At least from my first impressions (and I strictly am talking about first impressions for all of this, I haven't engaged with this much at all) the frequency seems a bit high.  My character is on spring 66 (so 6 days in) and got one portal storm (thankfully when I was inside and sleeping as it had no warning beyond when it started), and I just received what I assume is a warning from a second incoming storm.  With this frequency I don't know how much time I can actually commit to something and how soon I might have to bail.  Part of how STALKER maintained fairness with the blowout system is that the map is laid out in such a way that you always have adequate time to get to shelter -- that way, blowouts aren't something you need to actively worry about until you receive a warning.  It's harder to maintain that in a procedurally-generated world that wasn't particularly designed with this in mind, but that should be kept in mind when tuning warning time and frequency.

Obviously at first I naturally want to stay inside to avoid portal storms -- this is fine, but I foresee some potential problems from it.  Firstly, I don't really know when it'll become "safe" to explore the storm -- I typically play with some very generously statted characters (I typically go straight into the cities killing dozens of zombies with relatively little risk), but many of the effects I've heard about seem like they'd be crippling to nearly any character.  Maybe this could be mitigated by ensuring that you can "test the waters" and engage with the storm in a somewhat controlled manner -- as in, at least for low-level storms, you should always be able to retreat back inside with relative ease if things aren't going too well as soon as you set foot outside.

Secondly I'm a bit concerned about how this impacts different starts.  This'll probably have by far the most impact on forest starts where actual shelter may not be present for some time.  I'd most likely be dead already if I got a day 3 portal storm like I got on my current game, if I wasn't using an evac shelter start.  In order to be fair to players who start in the woods, I'd say a storm shouldn't happen until a point where you'd be very reasonably expected to have shelter (and with extra generous warning for the first storm), or as a potential alternative perhaps tree canopies should count as valid shelter -- in fact, this might be one of the best ways to ensure that cover is always available on relatively short notice.  

Finally, I am also not sure what to think of the idea of storms increasing in intensity and frequency, etc. on their own.  If I'm going to be retreating to the evac shelter for the first several storms, and they get more frequent and intense, will I ever know that I can handle dealing with one head on?  It might be better to have intensity and the like scale with your engagement with the mechanic -- as in, cowering in the evac shelter won't make the storms change much, but if you're out there wheeling and dealing with the extradimensional invaders and raiding their dungeons, naturally they might turn to pursuing you more aggressively.  Part of this is meant to create a threat the player has to deal with that isn't fully on their own terms, I know, but I don't think every character can be expected to match the challenge on the same timetable, especially if effects while indoors or being literally pulled outside are being considered.

**Comment by Light-Wave** [comment]
Got an idea for a potential 'carrot' for players to be outside during portal storms.
The alien grass could spawn with strange seed pods in the center that drops a collection of random seeds if opened / destroyed. That would have the added bonus of making all seeds available in the innawood mod.
Furthermore, it could be worth considering the idea of being able to hide in bushes or under trees. I also like the idea of being able to drape a blanket over yourself to hide from the monsters.

**Comment by kevingranade** [comment]
Something that bears some thought is what is it about portal storms that makes it where you can hang out inside a structure and avoid most of the effects.  Nailing this down might make it easier to reason about what effects are or are not going to be able to reach you (for example "the voices" definitely can reach you, but most of the monster spawning does not) and lead to interacting with them be more interesting. Likewise that would help inform whether you can do something about protecting yourself by means other than hiding in a structure.

Some ideas, none of which should be taken as definitive:
There's a *thing* that spawns somewhere, possibly somewhere in midair.  If you can see it or it can see you, you get impacted by the portal storm.
As above, but it's a portal far in the air, such that you are affected if you can see (or are seen by) the sky or a certain part of the sky.
For a less intentional/mystical version of the above, it could be some form of radiation, what blocks it?
The sky itself takes on the form of a cogitohazard that makes you vulnerable to the effects if you see it.
A gas or liquid (or something that acts like one) falls from the sky and inflicts the conditions on you, but can't enter structures, at least not rapidly enough to affect you.
Invisible tentacles flail through the are covered by the portal storm, if one touches you the effects trigger.

**Comment by Zeropol** [comment]
While portal storms definitely add some challenge and uncontrolled hazards to deal with, I agree with some of the above comments that it could be optional.

Because it walks on kinda paranormal grounds that shift the feeling of playing the game beyond the already unbelievable state of multiple dimensions and aliens. ( already in lore and I'm not fond of. I wont argue and agree zombies need some justifications. Human made just looks more plausible to my narrow mind and it became harder to RP )

I of course played tens of hours with this addition before talking, and in no way pretend my vision is good. Just a player feeling. I love the added challenge to seek shelter every now and then ( I question the innawood play in the actual state but I'm sure it will be balanced eventually ). 

I would love environmental hazards, expending the weather system ( is this forbidden to talk about acid rain back ? ) out of scope of this PR yet using the EOC, that force the player to seek shelter, without involving SCP.

As a constructive message, balance wize, I would share the feeling that portal storms spawn too much opponents, have a too long warning period ( sometimes 5+ hours ) and dont affect NPC.

Absolutely not a flamebait message, if it is considered as one, feel free to delete it.

**Comment by Terrorforge** [comment]
I agree about the warning messages. I didn't like when the storms would come out of absolutely nowhere, but I _did_ like that they sent me scrambling for shelter on short notice. Now that the warnings are unmissable, I think it would be fine to significantly reduce the time between the first warning and the start of the storm.

Perhaps it could be another variable that gets worse as the storms increase in intensity? E.g. the first time you get a storm you might have a full hour to find shelter, but after a season or two you only get minutes.

**Comment by gmlacrosse** [comment]
This looks like a pretty nice addition to me. I do wonder if it might be better if it's an optional setting and not default. Also, make the severity of the storms a world configuration value. There should be a risk/reward to it. If you decide to risk venturing out into the storms there should be something to be gained and not just a lot of headaches and frustration.

**Comment by NetSysFire** [comment]
Currently you even get attacked and hurt when you have the Debug Invincibility and Debug Invisibility and similar mutations. Would be nice if the weather respects that.

**Comment by tsuginiomaewa** [comment]
Looking at Kevin's comment saying to think of what it is about portal storms that makes going inside safe, how about leaning into the whole STALKER aesthetic portal storms have a bit more?

Specifically, in terms of the effects having an origin. I'm not saying for it to have a defined origin on the map (though that could be interesting), but like how secondary to emissions, STALKER has psi-storms (at least I know Call of Pripyat does). From what I've seen of them ingame, psi-storms result in the sky gathering a sort of energy, with audible crackling as it grows larger, before finally releasing it to the ground like a bolt of lightning, leaving a ringing echo even if you're indoors and giving a brief pause before doing it again. Contrast that to the emissions, which emanate a wave that advances outwards from the Chernobyl NPP (at least it seems to come from that direction).

How about a sort of combination between the two? Portal storms would result in "bolts" outside like a lightning storm, with these bolts being _extremely_ bright (enough to slightly illuminate through boarded up/sealed windows, and definitely enough to luminate past what the sun normally does and hurt the eyes of anyone able to see that flash). These bolts would radiate its effects outwards in a wave (which wouldn't _have_ to be modeled, but if doable would be cool to see the alien grass spread in a direction), and spawn enemies at the point of impact.

Provides the explanation for why being inside is safe (emanates from those bolts, it's not going to go through walls and solid objects), as well as creating a strategy of ducking in and out of buildings during a storm to dodge the effects, and using the lull between bolts to either get from place to place or grab whatever incentive the storms create for going outside.

**Comment by Zeropol** [comment]
> As to your balance thoughts those sound pretty fair and after I make nether dungeons I'll be making changes to those things.
> 
> I've thought a lot about other weather effects and would love to hear your thoughts, portal storms are among other things supposed to be an example of what you can now do for weather with no code changes. Making new weather stuff, or mods for a different flavor/spin on portal storms both sound cool.

Thanks for your kind answer.

As portal storms related weather effects, comes to my mind : 
- Suddenly in daytime it is night-level of luminosity / Suddenly in night time it is day-level of luminosity.
- Sounds that force the player to wear some kind of hearmuff to sleep, and miss their alarm, or delay sleeping time (may be considered boring by some players). At the moment, sounds from the storms don't seems to affect the ability to sleep.
- The lightnings suggested by Tsuginiomaewa is interesting, mainly for ambiance and alien grass spread. I have no real feeling about them spawning ennemies. Bolts doing damage to player and/or starting fire may be a bit unfair.

Not related to portal storms : 
- Ligntings that (can) weak up you at night ?
- Some kind of "soften" acid rain ? ( maybe only a greater mood debuf than rain, without pain or items deterioration )
- Shorts earthquakes ( player can stumble )

Thanks for your work and attention to players feedback.

**Comment by chrispikula** [comment]
So, now with the PR right before this, and the one where vehicles take permanent damage over time, is van life and sleeping inside RV's with all the windows curtained now effectively doomed?

**Comment by chrispikula** [comment]
I couldn't see if the PR actually set up a different spawn list, so I've got to assume it didn't.  But there are ones(attacks?) that try to pull you out, it seems?

**Comment by Ramza13** [comment]
At the moment none of the new portal storm only monsters can smell and in my testing they didn't make any efforts to bother me as long as I couldn't be seen.  That said some of the classic nether spawns can smell and I didn't think about that.  If that becomes an issue I'm happy to look into it as I've always toyed with having portal storms be new monsters only.  

At some point the idea of a monster smart enough to open doors or an effect that opens doors/windows is also a possibility.  This would all have to be balanced to be not super common/make portal storms shorter to make being inside still good if not perfect.

**Comment by chrispikula** [comment]
Ah, k.  Otherwise each portal storm is going to turn into tower defense.  :-p
(vehicle or base!)

**Comment by esarbe** [comment]
Portal Storms in experimental seem to create fine smoke indoors and around buildings. Is this intended?

**Comment by esarbe** [comment]
With regards to getting players out to explore during portal storms; how about having a powerful temporary buff that fades away with time so that players *eagerly await* the next portal storm? Make each buff the player had count towards difficulty gaining the next one.

**Comment by Ramza13** [comment]
>Portal Storms in experimental seem to create fine smoke indoors and around buildings. Is this intended?

Just submitted a fix.

> With regards to getting players out to explore during portal storms; how about having a powerful temporary buff that fades away with time so that players _eagerly await_ the next portal storm? Make each buff the player had count towards difficulty gaining the next one.

So my current (very vague) plan is to at some point enable a playstyle that is buffed by portal storms.  I'm not sure if I want it to be a mutation of some sort or artifact based or the result of bargains with nether entities but I do want at least a way to make some characters get buffed by storms.  I don't think doing it for everyone is a good idea though.  That said my ideas for this are still early brainstorms so if someone else has more fully formed ideas they won't be trampling on my toes.

**Comment by drhead** [comment]
Have you considered an alternate path where players can do things to mitigate portal storms with the help of other survivor factions (probably Hub 01)?  I've always wanted some sort of mission where the player can mitigate some of the effects of the Cataclysm, sort of in the same vein as the satellite launches in Half-Life that are used to close portals.  I don't think it'd be appropriate for a lone satellite to singlehandedly end portal storms, but perhaps something like that could give a 24-hour warning of when the next portal storm is to occur?

**Comment by drhead** [comment]
I have had some issues with portal storms within my basecamp.  It's my understanding that you are still intended to be more or less safe while indoors even with recent changes, but that isn't exactly the case anymore.  For reference, my basecamp is an evac shelter.

The first issue I have is that it appears that I am having incorporeal applied to me while inside.  I can't prove that it happens unless I am adjacent to a wall, I've had it happen very frequently when near a wall.  It's a great annoyance, especially when I keep rewinding away from my pile of clothes. (also a note on incorporeal -- is there an explanation for why we don't fall through the floor?)

I believe you've already been made aware of some of the older netherworld creatures being able to smell -- honestly that's not TOO much of an issue, at least a bit later in the game, I personally solved that by replacing the doors and windows with metal doors and reinforced windows with shutters, and honestly I kind of like the extra feeling of security!  But last storm, I believe I had an Absence break one of the walls while I was sleeping -- I am not sure the absence did it, but the absence was in there attacking my NPC.  Fortunately I woke up because incorporeal made me drop my earplugs.  They may have detected my NPCs by sound, since I forgot to set one to stay quiet, and he may have been commenting on the sounds outside.  Honestly, this could work either way in terms of safety, players could always make use of the evac shelter basement if this becomes a regular issue.  But I'd assume that if you're talking about a monster possibly opening doors or windows, that surely bashing walls down isn't intended?

Final note on basecamp safety -- you might want to look at NPC behaviors during storms.  I explicitly instruct my workers to be quiet, not investigate noises, and to not open doors, because I don't trust them to not get themselves killed.  But I think it'd be best to make NPCs automatically cancel all jobs, go inside, stay quiet, and not investigate noises during a storm, and preferably I'd want to be able to set a zone for storm sheltering so I can make them go in a specific building without worrying about their preference for hanging out in the forge.  Of course, the NPCs should also help warn you during the early stage by saying, "Portal storm soon, fellow survivor!"

**Comment by SenpaiSlime** [comment]
It had crossed my mind recently that there might be a thing we can do to help alleviate the current instant monster spawning from portal storms, for the sake of players. what you can do is replace current spawns with a temporary portal monster that would turn itself into another monster after a few turns. This would give players an indication that *something* will appear there. As far as the storms go I would personally prefer monsters either permanently spawning from the storms or not at all (minus the tendril one maybe, as that one feels very fitting for that exact thing.) and I agree with having your own roster of monster from the portal storms. And maybe slap on some spell effects so we can have fields of gas or lighting strikes or etc to make the storms more visibly chaotic. Id also recommend removing flaming eyes from the storm spawns if they are there currently, as themselves alone can be a death sentence to a player under normal circumstances. I have been loving the storm addition so far (yes i enjoy STALKER) and greatly anticipate further additions. I have a couple nether monster concepts on my back burner if you want an idea or two!

**Comment by Light-Wave** [comment]
Some more thoughts on the balance.
- Memories having grab makes them rather deadly with the new strangulation mechanic.
- Hiding in an improvised shelter or similar gives almost no protection right now, as enemies still spawn and come for you. Reducing their sense of smell, hearing and/or sight of spawned monsters might make innawood shelters more relevant.
- Making spawned monsters unable to damage buildings and vehicles could also help reduce complaints about that.
- It would be nice if portal storms could lessen in intensity as well, and if we could see what intensity the current storm is at (low, medium, high). That could make it so that end game you might notice that the current portal storm is weak enough that you're willing to brave it.

**Comment by AcidAntOnAMinefield** [comment]
> Reducing their sense of smell, hearing and/or sight of spawned monsters might make innawood shelters more relevant.

Imo the best way to "work" with this would also be to completely remove the ability of smelling for monsters that don't breathe, because realistically, not breathing and being able to detect scents makes no sense.

**Comment by chrispikula** [comment]
> Imo the best way to "work" with this would also be to completely remove the ability of smelling for monsters that don't breathe, because realistically, not breathing and being able to detect scents makes no sense.

Maybe tie it into a weakness tag, which would mean that only creatures with "BREATHING"
 can be affected by poisonous clouds?

**Comment by NetSysFire** [comment]
Also, someone in the IRC has reported that you get funny messages when ramming specific portal storm monsters with a vehicle (shame that @TaxyTails did not report it themselves):

```
Your Hyphen's |\ steel plating rams into the less than nothing with a smash!
```

That looks unintended.

**Comment by ProjectileWaterfowl** [comment]
I'm having portal storms start *while I'm sleeping* and having portal storm enemies spawn and actively tunnel through my fully roofed vehicle to get to me. Even becoming incorporeal, while in a barricaded room while under a roof. Is this intended?

**Comment by dseguin** [comment]
I think there might be an issue with what the condition considers as "outside". Might need to add a condition for enclosed vehicles as well:

```diff
diff --git a/src/weather.cpp b/src/weather.cpp
index c883fd39e2..130df12317 100644
--- a/src/weather.cpp
+++ b/src/weather.cpp
@@ -43,6 +43,7 @@
 #include "translations.h"
 #include "trap.h"
 #include "units.h"
+#include "vpart_position.h"
 #include "weather_gen.h"
 
 static const activity_id ACT_WAIT_WEATHER( "ACT_WAIT_WEATHER" );
@@ -73,7 +74,9 @@ static const trait_id trait_FEATHERS( "FEATHERS" );
 bool is_creature_outside( const Creature &target )
 {
     map &here = get_map();
-    return here.is_outside( point( target.posx(), target.posy() ) ) && here.get_abs_sub().z() >= 0;
+    const optional_vpart_position vp = here.veh_at( target.pos() );
+    return here.is_outside( point( target.posx(), target.posy() ) ) &&
+           here.get_abs_sub().z() >= 0 && ( !vp || !vp->is_inside() );
 }
 
 weather_type_id get_bad_weather()
```

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/portal-storms-tweaks-for-early-vs-late-game/27671/7

**Comment by MochaModifier** [comment]
> This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:
> 
> https://discourse.cataclysmdda.org/t/portal-storms-tweaks-for-early-vs-late-game/27671/7

Ah, that'd be my post sorry - oddly, I didn't find my way here first via google, and I'm just starting to interact with the community - my apologies for any redundancy.

I would sum up my feedback as - weather shouldn't be out to get the player specifically; and weather that can directly kill the player with one guaranteed solution isn't to fun. Being yanked out into it wouldn't endear it to me as a fun or fair challenge.

Constantly spawning enemies, especially powerful ones - seems out of scope for a 'random weather event that can happen anywhere, with some regularity"; and it feels like right now it happens more often than most any other weather. I can't remember the last time it rained in game.
 
As it is now, this should be a rare event, or one that happens in specific locations/regions only. But with tweaks, it could easily be adapted as an unusual/uncommon weather - not a pressing danger or uber threat, but an interesting modifier on the day to day which could be helpful or harmful given the situation. Several strengths of the storm - not in terms of what it spawns, but general effects, could work. Literally, a Portal Drizzle, Portal Storm, pehaps a Portal Monsoon.


Doesn't seem to out there that these portal storms are an after effect of the cataclysm (obviously); but the reason why also exists in game I feel. 
After the government tried to shut down, control and contain what happened - doubtless this included the sources of the portals/portal storms: Labs working on portals. Not every containment effort was a success - and some places remained 'open' to some extent - and act as the source of all the storms. These are strongest above the labs, and get weaker the further they go - hence the different strengths.

As for the carrot, I definitely think of Roadside Picnic, or STALKER. But I also consider that Blowouts only happen in The Zone - where artifacts are found. This could be the set up for such a zone.

Other Labs could serve as a source of protective gear as well - as I understand it, the scientists were messing with other dimensions and portals for some time - and surely they'd have some kind of HEV suit, right? 
Survivors might later on develop special suits a la STALKER, which have environmental or anomaly resistance. People might even develop sensors to help deal with storms - _or find that carrot_.
I'm not sure if the game has this yet - but survivor factions could at some point begin producing protective gear based on their experiences post-cataclysm.

I propose this with various issues in mind:

- Storms currently vary in threat depending on not just time, but the kind of start you pick. This can cause balance issues with these scenarios
- It's very hard to understand or grasp what the storms are, or how to tackle them 
- Only some character builds may be viable to survive a shelter-less start, or if not hunkering down (which affects the balance of many trait, stat and skill choices just for a weather effect)
- There's no obvious reason not to hunker down, outside it being inconvenient/slowing play

Issues like them happening to often, I think only come up as it exacerbates the above issues. The frequency may still need to be toned down - as a special feature, it feels like it occurs more often than rain, fog, even sunny weather.

Grounding some aspects of the portal storms in reality - and I mean this in the sense of "CDDA's reality" and not ours, I hope will enhance the nature of portal storms and better fit it in the gameplay and design of the game.

One example I might offer, is that the game has ways to prepare for just about any hazard or threat; and try to up your chance of survival. These are often fairly reasonable if not obvious - armor to prevent bites, ranged weapons to not get boomer bile on you, thorazine to help when experiment with shrooms etc. etc.
Perhaps the creatures being spawned don't in fact exist - but the portal induces hallucinations by releasing some chemical to the environment - once the player understands this, the storms could then be a threat which requires preparation. These things act very similar to hallucinations - and either a leap of logic, or a survivor note could help inspire the decision to wear a mask and rain coat.
These might not be good as an HEV suit - but things like thorazine, face masks, raincoats - these could give some resistance or stave off the effects to allow going outside; while giving you room to add better carrots.

**Comment by chrispikula** [comment]
I've vaguely understood that the existence of Portal Storms references in-game are a direct reference to the Half Life series.  So they were worst right before the player takes control of the PC, and are a bit quieter now.

Side note, shouldn't this be tagged as a release blocker?  Given that since Dec 9th we've been in project freeze and this started before.
#53318

**Comment by remd** [comment]
I use auto Safe Mode so this may not apply if you toggle Safe Mode off for the duration of a storm. I'm not sure.

When Portal Storms spawn new monsters close to the player it can be frustrating trying to craft, read, or otherwise utilize the time indoors waiting out the storm. New monsters spawning in will start causing warning messages, e.g. the "Monsters spotted! Stop crafting?" message or "The Memory is dangerously close! Stop moving items?" and you are forced to constantly "Ignore further distractions and finish". Since monsters phase in and out of existence you are repeatedly prompted when new monsters appear. This is all in spite of the fact that none of these monsters can affect you so long as you stay inside and don't open any entrance.

I sort of enjoy that the storms "distract" the player with beckons for help, strange noises and phrases, etc. but this doesn't feel ideal. I've even positioned to block LoS while retaining light for crafting/reading or worn earplugs to avoid seeing or hearing the monsters. Maybe that's somewhat intentional (especially wearing ear plugs to block out the sounds of people crying for your help) but once your ear plugs are in and you've given the "Ignore" instruction once it's no longer fun for the Portal Storm to distract you from what you are trying to do. 

Sure, decrease my focus, slow my crafting by making it dark outside (it's odd that a Portal Storm does not interfere with sunlight reaching the ground), but time should just pass once I've decided to ignore what's going on out there.

Consider making the monsters neutral until they have a clear path to the player (might be too costly to calculate), or the player is considered outdoors. At the moment the player has to turn off safe mode or ignore each new monster just to stay inside their own base for the day. Something about that feels off to me.

**Comment by JLubbs** [comment]
Very tedious after awhile. I really wish they had a way in Worldgen options or settings to disable.

Currently manually killing all PS via JSON modification.

If there is any sort of reward for having them it is very unclear.

**Comment by AcidAntOnAMinefield** [comment]
> Very tedious after awhile. I really wish they had a way in Worldgen options or settings to disable.
> 
> Currently manually killing all PS via JSON modification.
> 
> If there is any sort of reward for having them it is very unclear.

Yeah, I agree. I think it's one of the features that really needs a toggle on/off option. I've played a lot through them until I stopped playing the game about 3 or so weeks ago; the only thing the Portal Storms are "good" for is to slow the player's progress to a grinding halt - or to terminate the player.

**Comment by Marrim** [comment]
I tried this for a while now, half a year into the game and it forces you to be always aware of nearby shelter in case you need to duck. Gives a strong Stalker vibe. And not all shelter is viable, anything with breakable windows will not cut it especially if there are zombies nearby. I've also had twisting blades spawn inside my car (presumably because quarterpanels prevent walking in but do not make the car truly enclosed), and even though I was able to hold out, a basement is much preferable.

Overall, I like the mechanic very much, though the storms happen a little too often for my taste. And I think special consideration needs to be given to how they interact with NPCs and basecamps.

**Comment by esarbe** [comment]
During my last visit to a Refugee Center, a portal storm started. I tried to wait it out, but then all the NSCs ran out to fight the monsters, many of them got killed. :laughing:

**Comment by Ramza13** [comment]
Yea adding a bit to animal and NPC AI to handle portal storms is on the to-do list, I haven't worked on AI at all yet so I'm not sure exactly what it will look like but its being thought about.

**Comment by esotericist** [comment]
so, been meaning to post here for a bit, but life intervened. alas. also, i have to admit to only having skimmed the above comments (there's quite a lot here), and i am late to this because i was staying on stable until relatively recently. that said, i have a whole novella's worth of thoughts, but i don't think i can quite get them all organized yet, so here's what i've got to start.

the biggest issue with the portal storms so far, in my opinion, is the whole issue of counterplay. for a span of content which is substantial enough it functions as an entire subsystem of a game, it's very important to ask questions like: what are your options? what do they look like? what are the consequences of those options?

at least in their current form, portal storms offer one option: hide. there are no meaningful alternatives. you *can* fight, but it won't achieve anything except burn resources. you can't run. you can't erect any kind of meaningful defense. you can *only* hide. and given that's the only option you have, by necessity, *it must work*. a gameplay subsystem which only presents threat and no meaningful response at all is a massive attack on player agency.

notably, right now hiding only *sorta* works, and some of the stated plans above clearly intend to reduce the efficacy of it. this, i feel, is a substantive problem. as long as you cannot fight or escape a portal storm, it *needs* to be possible to successfully wait it out. this means no forcible teleportation if you're trying to hide and doing so in a sensible fashion (i.e. completely out of sight, completely unexposed to outside). this also means no enemies magically knowing your location and tearing down terrain to get to you. more on that later.

if more meaningful options can be provided, having more nuance between the types of consequences of different counterplay options makes sense; if a player has to pick which thing they can live with, that's fine, as long as they *have* a meaningful choice. we're not there at this time, though. in the current form, to many players, portal storms are a "you basically have to stop playing for a while, and you might get your run unfairly ruined anyway even if you do everything right" event.

the next big thing that's important with a subsystem-scale content span like this is how it interacts with other systems. i see two core issues with this right now:

1. it's already been discussed that NPCs don't know how to deal with portal storms, and obviously NPC ai is a whole huge herd of unshaven yaks at the moment. that does not mean, however, that it's appropriate for portal storms to be able to break our NPC-centric content just because the NPCs aren't smart enough. we need better safeguards to keep portal storms from being as severe until the NPCs can react intelligently (or we can at least *fake* them acting intelligently); this means the portal storm events shouldn't be targeting NPCs the way they can the player, since NPCs are unable to formulate plans and enact them. they can't engage in counterplay even if they have options, so the game shouldn't ask that of them right now.
  * * notably, i've noticed a tendency amongst some players and contributors to equate portal storms with other threats to faction locations. i believe this is to be a substantive mistake, due to the levels of agency involved. zombie horde trundling towards a faction location? they can be steered in another direction (this even works with the currently incomplete in-game implementation of wandering hordes, although it is exceedingly tedious). mycus or other terraforming entity creeping towards a settlement? set up firebreaks, regularly prune edges. even ants and other infestations have something players (and ideally eventually npcs) can proactively do to protect their territory. failing any of the above, you would have time for you (or the faction, not that they can decide to do this yet) pack up and move to a less vulnerable location. portal storms have no such thing: they happen when they happen, where they happen, and there is nothing you can do except not be there. and you cannot, in advance, know not to be there. this makes them qualitatively a different category of threat than all the other mass dangers.
2. the events like to emit loud sounds at the player's (and presumably also NPCs' locations, based on some of my negative experiences in #56593 ) location. this interacts *very* severely badly with how critter movement and bash decisioning work; as of #44751 critters that can bash will bash if 'sufficiently interesting' sounds are heard, and speech is in the categorization for 'provocative'. the end result of this is that every time one of those yells is dropped on the player's location, anything that can bash (and this includes a number of the portal storm spawned entities) is being tasked with mining their way to the player's location. the necessary fortifications for dealing with this are nontrivial, and mostly the 'solution' here for players is to go underground. this needs to change.

now i'm going to take a moment to apologize for the wall of criticism above. it needs to be emphasized: i love the idea of the portal storms. i adore a lot of the writing in it; the messages are very well crafted, and the sense of forboding they provide and the alienness in them is in my opinion a very solid addition to the overall c:dda experience. the last thing i want is for anyone to think i don't want the portal storms to be in the game; they belong here, and i'm actively looking forward to future content additions (such as the portal storm dungeons; i love the idea of having both the ability and *reason* to go diving into one).

so i feel the storms need to stay, but i do think before we can consider a stable release containing them, we definitely need to address the problems. for that, i recommend the following, or something that accomplishes the same goal (i'm not wedded to specifics here; i want the game better, which means there are things which must not be, but i don't want to dictate the exact form it must be.):

1. the sound-based events dropped at the player **need** to be moved to another type, or otherwise somehow excluded from monster provocation. the volume level might also be an issue, since i believe monsters can react to volume independently of provocation type. this is a high priority item.
2. forcible teleports need to be always limited to players exposed to outside, and should not happen until substantive stages of escalation
3. npcs need to be exempt from the majority of things that happen in portal storms. this can be stepped back once the npcs are able to exhibit agency (or the illusion thereof), but as long as the npcs are rock stupid, we can't let the player's presence cause death of npcs for things that cannot be in any way mitigated. the simplest solution in the short term is probably an exclusion zone around faction locations; portal storms simply cannot start while you're within a specific range of them (but perhaps could start and still continue if you stubbornly run *to* the location once they're already in progress)
 * * related, player faction camps should also probably be protected, as should situations such as @I-am-Erk's work in progress at #56629. arbitrary locations with random npcs might not necessarily need to be protected, but any npc that is part of a faction quest chain or any other significant content linkages needs to not be subject to being broken by what is essentially an arbitrary dice roll.
4. related to #56593 some of the effects could probably use some tuning to mitigate their ability to do strange things. for instance, teleport effects should probably not be able to act on immobile critters (they are generally conceptually *objects*, and are only creatures out of technical necessity for game implementation)
5. i think we need a slower ramp-up in portal storm severity; perhaps some earlier prolonged stages where things are merely 'weird' and not in any way directly attacking the player (although still presumably putting strange effects on the player from exposure to outside). this will allow players to be aware of the portal weather, without being directly threatened... and might help make it easier for players to decide not to immediately hide once things start to escalate. this could greatly mitigate the "i have to stop playing when one of these pops up" conclusion currently experienced by so many players.

longer term (i.e. this doesn't need to happen before 0.G), i think for the more severe states of portal storms it might be nice to be able to see them moving on the overmap (similar to wandering hordes), as a visible weather pattern you can actively seek to evade, or otherwise seek a truly well fortified location. portal storms of that grade would make more sense to attempt 'siege breaking' activities, but to reiterate, this should a) be very advanced in the progression, b) something that can be *actively fled from*. ideally you should still get a warning when they become visible to the player, as a prompt for them to look at the map and decide where they need to go, and perhaps attempt to guess at how much time they have to pack before they do it.

**Comment by MylieDaniels** [comment]
I would like to say my primary concern with portal storms is the direct lethality.  A lot of things have been said about portal storms primarily not doing direct damage, but uh.  I find this to be a bit incorrect.  Monsters that put you to sleep and exhaust you are massively lethal when paired with monsters like the shrapnel swarms and giant appendages that can kill you "normally".  In addition, screaming at your location beckons all monsters to you, which in turn causes death.  I love portal storms, but I think the consequences of combining their effects with both their other effects and the rest of the world needs to be more heavily considered.

**Comment by danielkwinsor** [comment]
This feature does not match my playstyle.  I need to have a toggle for this to turn off in worldgen, or this is going to be the end of playing CDDA, sorry.  Short feedback: very tedious, happens too often, no reward and the penalties I know are going to just get worse over time.

Alternatives to an on/off switch are to give a worldgen setting of allowable distance from an actual portal for this to trigger, or to provide an in-game object that must be interacted with that starts a chain of events.

But, to leave it with no option to turn it off will be a game ender for me, sorry.  I know it's a nice feature with personal effort, but this is the antithesis of my play style.

**Comment by esotericist** [comment]
there's no need for a setting for this; one can make a mod that disables the portal storms. we also have scenarios that alter the frequency and severity of portal storms in the core game as examples of some of that functionality.

**Comment by danielkwinsor** [comment]
Please provide a mod in the default game.

**Comment by esotericist** [comment]
> Please provide a mod in the default game.

that's not compatible with our policy; we don't provide discrete option mods, especially ones that simply turn off a feature we intend to be core gameplay, as this has been consistently proven to *inhibit* work to make controversial mechanics work more smoothly as people simply switch it off rather than work to make it better.

now, if a substantive content mod that qualified for in-repo inclusion were to incorporate such a thing, that would be different. (it even seems likely; dark days of the dead is a good candidate for this). but we're not going to provide a "hit this switch, portal storms go bye bye" function, regardless of form. this keeps the feature from being refined.

**Comment by danielkwinsor** [comment]
I've given 2 alternate options.  But seems like most likely I'll be spending my playtime elsewhere.

**Comment by esotericist** [comment]
that's a choice you're free to make.

**Comment by SariusSkelrets** [comment]
> Short feedback: [...] ,no reward 

Wait for it. Portal storm rewards are coming soon. A PR of mine #56566 adds a way to get useful stuff from the storms

There's also a Hub 01 quest added by #56646 that can give you gold coins just for standing outside in a storm with their device

**Comment by I-am-Erk** [comment]
So I spent some time thinking on @esotericist's concerns last night and have a couple options.

1. My personal favourite, would require redoing the lore a bit. **Portal storms are an individual experience**. Think of it as "going out of phase with your dimension". Only you can see and hear things in the storm, they can only affect you. We take out the ability of storms to move and teleport things permanently (they could temporarily remove a chunk of the map and then it would go back to how it was after). NPCs don't react to them and can't be hurt by them. Later, we use EoCs and scripting so that your NPCs can get locked into their own personal portal storms, which you can't see and have trouble helping them with. In this version, we can then have these storms become worse not so much over time, but as you "mess with reality". So, getting teleglow for example would permanently add a small step to your portal storm frequency and severity counter.

In this version storms might still be something you can see coming, since what you'd perceive would be exactly the same... it's just that it's happening to you, not to the world. However this does lock out some possibilities for what the storms could do later on.

2. some careful auditing of storms as they exist.
- Agency: Players need some ways to influence storms. I would suggest that, similar to my "mess with reality" idea above, we have storms on a much slower timer but your personal actions can speed that up by doing things like killing floating eyes, activating artifacts, and playing around with tears in reality. If you choose not to do these things storms still advance (we assume NPCs are doing this stuff off-screen) but it happens much slower.
- AI: For a start I suggest that, even if storms are 'part of the world', we make the portal storm sounds only audible to the player.  This will stop them screwing with monsters and NPCs quite so severely. We should take out the monsters that make a ton of noise. That goes down into Danger.
- Avoidance: Going inside should make you safe. That is a fun aspect of the storms we shouldn't lose. Once your doors are closed and your earplugs in, I think that should be all it takes unless you've ramped up storm danger later on. Even if you see things outside the windows (which you should) they should be harmless, possibly not even triggering Safe Mode, just scary for noobs. NPCs should give advice about wearing earplugs too. The EOC that causes portal storms should trigger NPC AI to run for shelter and stay there.
- Danger: We really need to think carefully about the monsters spawned in a portal storm. I'd like to suggest that, by default, monsters should not spawn at all. We could spawn some purely hallucinatory creatures, that would be really fun: things that are out of phase with our dimension and can be seen, but not interacted with. However, portal storms spawning creatures that can hurt you (by damage or secondary effect) should probably be kept to a *plot event*. Then, we should start writing up story line stuff that strengthens the storm and makes it possible for creatures to come through physically, creating some of the bigger problems we currently see. This could be triggered through several routes: I suggest having hub01 portal related missions, exploring XEDRA portal facilities, or following directions given to you by the mysterious portal voices. It might be nice if, once opened, this Pandora's Box also provides some of the proposed portal benefits, and eventually wears off on its own (perhaps lasting longer each time you do it). It is established in the lore that portal storms like this were a temporary part of the early cataclysm so it's safe to assume they naturally revert to a weakened state.
- Dimensions: We should play on this "alien grass" trope further. I would love to see the major portion of portal storms being that they run mapgen updates that temporarily turn segments of the map into other things, converting trees to alien trees, walls to creepy spiral architecture, etc. We will need to store the original contents of the chunk, and have it revert completely when the storm ends. Anything under a roof should be exempt from changing.
- Factions: NPC factions need to have a way to avoid the storms. In the short term I suggest we just add hidden effects to these areas to prevent storms from forming there. In the longer term we can add lore reasons once it makes sense for them to get them. Players also should have access to whatever they use to prevent the storms. Actually, lorewise and maybe mechanically, we could just establish that having more living people in a single place reduces the storm risk, so factions and faction camps are safe.

Is there anything that wouldn't cover?

**Comment by esotericist** [comment]
that seems reasonably comprehensive for addressing my concerns, at least. i have no firm objections to any of what you suggested.

as i mentioned in private discussion, i'm personally more interested in keeping the portal storms a collective event, but the personal event version has merit, and i wouldn't be upset if we ended up transitioning to that. it's a pretty big lore shift, though, and would have a lot of repercussions on what happens with the portal storms gameflow-wise. if you choose pursue that, it'll require adjusting a lot of the content plans that had been targeting the previously established collective event context.

as far as the creature-esque physical threats, the idea came up of leveraging the hallucination code, so they can be seen but not do much. then we can reuse the entities as actual dangers in our dimension-related dungeony content?

i'm torn on the moving-in-groups idea. it's currently hard to successfully transport NPCs due to their behaviors (as far as i'm aware we *still* cannot instruct a follower to stay and wait in the car without instructing them to guard their location, which does not prompt them to *guard the car*, but their world position instead). i'm not opposed to it in abstract, but i don't think we can benefit from that at this time without some clever thing to make it so you can have NPC tagalongs who are not desperate to be combatants, and do not follow you out every time you bring the vehicle to a stop.

**Comment by I-am-Erk** [comment]
On the player facing side I think the main place the NPC tagalongs would be nice is in your base, where you may have a few of them hanging out, and in random encounters, where generally you're going to be running into a small crowd of stationary NPCs who thus won't be getting messed up by sudden spawning portal storms. I agree they won't be quite as much help when travelling, although there may be some passive benefit

**Comment by esotericist** [comment]
sure, i was just reacting to the 'travelling in groups' concept. right now, much as i *want* to -- i like having an NPC in my car even if they aren't doing anything, part of why i consistently make sure my vehicles have multiple seats -- it's unfortunately awkward.

maybe someone can find a nice convenient shim for that.

**Comment by I-am-Erk** [comment]
Worth posting another issue... We could maybe have a "wait in the car" command

**Comment by I-am-Erk** [comment]
These are the results of a very productive conversation with Kevin, BombasticSlacks, esotericist and a few others the other day. I will see about getting each of them posted as an issue.

### Recommended portal storm course of action

Order of items does not necessarily reflect ease of implementation but none of these should be individually difficult.

#### Major Fixes
1. Make portal storm sounds into hallucinations that only the player can hear. Eventually NPCs should have a script to react to portal noises, but for now they should just be unable to react at all. This will prevent monsters randomly bashing things, and static NPCs leaving their zones to die.
2. Steal our current "wandering hordes" mechanic and use it to represent the "epicenter" of a storm. This allows it to show on the map and allows players to try to run away. Keep aspects of the horde mechanic such as following sounds, as this suits a portal storm well. Once the storm settles on a location (ie. lands on the player) it is assigned a duration based on its intensity, and once that duration ends the storm fizzles out.
3. Each time a creepy voice says something while you are inside a building, drop the duration of the storm. This keeps storms from uselessly persevering once you're inside and safely ignoring them.
4. Make storm effects start more gradually, with no consequences for being caught in an early storm.
5. Storm intensity/frequency should not rise as a function of time. Rather, it should rise as a function of "irritating the nether". Create a global "Storm Intensity" variable that is increased every time a netherum creature dies, or when an artifact is activated. Individual netherum deaths or artifact activations shouldn't do much but a few of them add up. This Intensity variable falls when a storm fizzles out, allowing players to burn it off if it gets too high. Teleglow temporarily bumps the Intensity score for a while (a few days), and going outside in a raging storm also causes an intensity bump for a few days, effective immediately (You hear a slimy voice call, "Ah, I see you now.").

#### Longer term fixes
- Many netherum monsters should be seen mainly or only during storms. In low intensity storms, benign nether creatures should show up, and in high intensity storms we start getting floating eyes and flying polyps. Moving these monsters to storms will make them more unique and compartmentalize some of the weirdness of our game. Killing netherum monsters spawned by the storm should *decrease its duration* but *increase the global intensity* stat, so that future storms will get worse because you fought your way through this one.
- artifacts should attract storms and increase their intensity, just a bit, so a stockpile of them causes problems.
- At first when monsters show up in storms they should be harmless ethereal versions of themselves. Later, the real thing should come in as the storm intensity rises.
- Add an Appliance that can be activated to cause a storm to immediately dissipate. Hub01 can design one after you do their portal missions. The Exodii may also be able to send you on a mission to get the parts they need to make spares of their own version. This Portal Antenna has to be built outdoors and has to be at least 5 tiles away from blocking terrain and roofs. In order to make it work you have to calibrate it up close which takes a little while (speed dependent on Computers skill) - creates a tense scene where you have to run out in the storm and activate the antenna, hoping you can get it running before the storm reaches dangerous intensity. While this drops the storm duration to 0, it does not allow the global intensity to fall, so if you want to lower the intensity of storms you have to face them.
- Give each storm a randomly determined "voice" that decides what messages it gets, and potentially what creatures and effects it spawns. Have the messages change with storm duration so that you get feedback that staying inside is causing the storm to fade away.

**Comment by Terrorforge** [comment]
I was just thinking about how my most enjoyable moment with portal storms have been when they surprised me, but at the same time you can't just spring them on players with no warning whatsoever.

What if you had a pre-storm weather, but that only meant there was a _chance_ a portal storm would start? Giving you a choice between staying home safely, and going out and doing your usual stuff with a _risk_ of having to duck for cover instead of being forced inside. This weather could potentially last a long time, like several days, possibly with several shorter storms occurring during the same "pre-storm". Perhaps some of items etc. could allow you to force this kind of weather to instantly become a proper storm.

Or you could take that to its logical conclusion and rework portal storms so that a "portal storm" is an ongoing low-level event with only minor distortions, accompanied by a chance of "portal flashes" that summon monsters, teleport you around, etc. The chance of flashes would ideally not be entirely random, but escalate and reduce in some observable way. Like a thunderstorm; you'll get drizzle and maybe some distant thunder at first, that evolves into a downpour and increasingly frequent lightning flashes until the rain becomes a deafening torrent, the trees scream in the wind and the roll of thunder never seems to let up.

Either way you could also add (or edit) a starting scenario where the world is always in this "pre-storm" mode, for masochists like me.

**Comment by jokermatt999** [comment]
I think these are a really cool feature that does a great job of adding more cosmic horror flavor to Cataclysm. However, I noticed portal storms still seem to be deadly for NPCs, as per [#58023](https://github.com/CleverRaven/Cataclysm-DDA/issues/58023). I really appreciate all the work that's gone into these so far, just wanted to mention it here because it doesn't look like you (Ramza) got tagged in that issue.

**Comment by Tharn** [comment]
I had this in another thread but I'll post it here, hopefully more concise. I'm also overstating some points for desired effect:

The big issue is that Portal Storms change the base game to a large degree. You have a playerbase that knows, plays and enjoys CDDA (evolve - gather - kill and loot - enjoy downtime - practice skills - murdercars - etc). Now you hit them with an unavoidable, binary, all-other-gameplay disrupting event that happens more than once a week. Classic CDDA is zombies, Portal Storms are supernatural Lovecraft magic type stuff. Yes, we have plenty of supernatural and sci-fi lore in the game now, but CDDA was always a `zombie roguelike`. You could say that this is the direction, and fair enough, but you may as well be saying that you're out looking for a new playerbase.

Let me describe it in a way that it was already described as earlier in this thread: I was intrigued playing through my first portal storm. I found the site, went into the dungeon and looped through it 5 or 6 times, wondering if anything would happen if I repeated it - if the voice would tell me something interesting or whatever. But it would just repeat the dungeon. So I went back out and got a reward. Fair enough. The second portal storm happened only a few days later. "Again?" I thought. I went into the dungeon, did it once and came out with a buff effect. I wasn't sure if I was supposed to do it like this, or *what* I was actually supposed to do. What *can* I do during a Portal Storm? It would stand to reason that you go into the Dungeon to "defeat" the storm. Make it stop. Make it happen less. Make some progress and understand what it's all about. But no, it just keeps happening, and it feels like from the 5-18 days window that Portal Storms are supposed to happen, they happen pretty regularly at 5-day intervals. From the third time onwards, it just became an absolute nuisance that I had to deal with. It was completely separate from my game of CDDA; didn't add anything but tedium and interrupted the game to a large degree.

If you are set on keeping Portal Storms, they should happen every 6 weeks *at most*. Give us the ability to fight this thing and avert the storms. Think long and hard where this new core feature works alongside the game and where it hurts the whole rest of the game that we enjoy playing.

I understand that, lacking wandering hordes, once the player has liberated a portion of the map and set up his base there, zombies rarely wander in there. This is your way of putting danger into pacified areas of the map. Personally, I would rather have actual hordes back in the game.

Small point: Storms as largely hallucinatory events would be a cool spin that I'd be much more in favor of. They could still wreck you when you're out dealing with zombies. They'd also sufficiently represent the physical world of earth gradually slipping into another dimension. Something as trivial as auditory hallucinations happening randomly at night not related to storms would also be a neat way to increase the spooky vibe without combing the whole game in another direction.
